goclone

```diff
@@ -76,39 +81,50 @@ done
 exeargs=$*
 
 # is github_user/github_repo
-if [[ "$src" =~ ^([A-Za-z0-9-]+)\/([A-Za-z0-9-]+)$ ]]; then
+if [[ "$src" =~ ^([A-Za-z0-9_-]+)\/([A-Za-z0-9_-]+)(\@([A-Za-z0-9\._\-]+))?$ ]]; then
```

Comment:
github usernames can't have `_`.

I see the branch name restrictions are little bit more restrictive, but your implementation should be okay for most cases: https://stackoverflow.com/questions/3651860/which-characters-are-illegal-within-a-branch-name

Reply:
Ah, looks like I incorrectly allow branch names starting with `.`, names ending with `.lock`, and names with `..` in them.

But, I am not sure I agree with being so aggressive in the regex anyway. I think UX is better if this is done as a check after the main parsing. For example, I spent a while thinking I'd regressed something when testing with `repo_name` only to realize the underscore was making it fail to land in that branch at all.

Reply:
Such illegal branches wouldn't exist in the first place, so checkout would fail. However, there might be some characters we're excluding, such as `/` actually is used somewhat commonly in the branch names. Maybe we should just do something like `(\@(.*))?`